src/sas/qtgui/MainWindow/GuiManager.py

```diff
@@ -206,7 +205,6 @@ def addWidgets(self):
         self.DVCalculator = DensityPanel(self)
         self.KIESSIGCalculator = KiessigPanel(self)
         self.SlitSizeCalculator = SlitSizeCalculator(self)
-        self.GENSASCalculator = GenericScatteringCalculator(self)
```

Comment:
`self.GENSASCalculator: "GenericScatteringCalculator" | None = None` instead? - keeps all variables declared in __init__ (or functions called from it at least)

Reply:
This and also not reinstantiating the GSC but reusing its instance. Fixed

Reply:
We have it already. `self.GENSASCalculator` lifetime is that of the application. Calling `show()` just makes the widget visible, while retaining the state.

Reply:
@rozyczko I don't think we do address what I was getting at (I should have said `closeEvent` btw, not `onClose`) - it needs to be set to not destroy the C++ side when the close signal is sent (apparently, this is dependent on Qts whims), as opposed to the issue with garbage collecting things on the python side. I've had bugs from this before, they we not fatal, but lead to things breaking in a subtly weird way - it's why the acknowledgements dialogue gets explicitly removed and re-instantiated.

Problem is python keeping alive something that Qt thinks should be dead, rather than python deleting something that Qt thinks should be dead.

Reply:
You are right, `closeEvent` was reimplemented in other classes in this file, not the GSC class itself. Fixed.